cli/silence_update.go

```diff
@@ -0,0 +1,147 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/prometheus/alertmanager/cli/format"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+type getResponse struct {
+	Status    string        `json:"status"`
+	Data      types.Silence `json:"data,omitempty"`
+	ErrorType string        `json:"errorType,omitempty"`
+	Error     string        `json:"error,omitempty"`
+}
+
+var updateFlags *flag.FlagSet
+var updateCmd = &cobra.Command{
+	Use:     "update <id> ...",
+	Aliases: []string{"extend"},
+	Args:    cobra.MinimumNArgs(1),
+	Short:   "Update silences",
+	Long:    `Extend or update existing silence in Alertmanager.`,
+	Run:     CommandWrapper(update),
+}
+
+func init() {
+	updateCmd.Flags().StringP("expires", "e", "", "Duration of silence (100h)")
```

Comment:
I'm thinking we should rename these. Maybe `expires` should become `duration`, with `d` as the short flag?

Reply:
I was just following the parameter names in `silence add` command here. `duration` does make more sense than `expires` though, but if we are going to change we need to update `add` command as well. That will be a breaking change though.

Reply:
good point